src/main/java/homework/task_8_exceptions/ArrayValueCalculator.java

```diff
@@ -0,0 +1,23 @@
+package homework.task_8_exceptions;
+
+public class ArrayValueCalculator {
+
+     int doCalc(String[][] arr) throws ArraySizeException, ArrayDataException {
+
+        int sum = 0;
+        for (int i = 0; i < arr.length; i++) {
+            if (arr.length != 4 || arr[i].length != 4) {
+                throw new ArraySizeException("ArraySizeException: Array should be 4x4");
+            }
+            for (int j = 0; j < arr[i].length; j++) {
+                try {
+                    int value = Integer.parseInt(arr[i][j]);
+                    sum += value;
+                } catch (Exception e) {
+                    throw new ArrayDataException("ArrayDataException [" + i + "][" + j + "]");
```

Comment:
What I recommend here - is to overload constructor of ArrayDataException with additional parameter Trowable and create ArrayDataException with message and Exception from catch block do not loose original stacktrace of exception

Reply:
Your comment has been taken into account.
The ArrayDataException constructor was overloaded with an additional Trowable parameter.